homeassistant/components/zha/helpers.py

```diff
@@ -922,9 +924,7 @@ def __init__(self, hass: HomeAssistant, gateway: ZHAGatewayProxy) -> None:
         hass_path: str = HOMEASSISTANT_PATH[0]
         config_dir = self.hass.config.config_dir
         self.paths_re = re.compile(
-            r"(?:{})/(.*)".format(
-                "|".join([re.escape(x) for x in (hass_path, config_dir)])
-            )
+            rf"(?:{'|'.join([re.escape(x) for x in (hass_path, config_dir)])})/(.*)"
```

Comment:
This is more readable than the version in [homeassistant/components/system_log/__init__.py](https://github.com/home-assistant/core/pull/125945/files#diff-d7d18627b016b0d21ad36d10f06b43146746bf30e77d78ecf7851c579f6795f1), I personally would replace the above version with this.

Reply:
I think it is actually the other way arround. As you read the regex you start seeing that it tries to capture some `x` that then it is stated can be either 2 paths given in the tuple. While if you remove the for loop, you can easily understand that is trying to capture either the `hass_path` or the `config_dir`.